Compiling Shape Regulariztation

## Issue Details

I've done nothing but trying to compile the code [Shape_regularization/regularize_simple.cpp](https://doc.cgal.org/latest/Shape_regularization/Shape_regularization_2regularize_simple_8cpp-example.html). I got an error message stating that the method regularize_segments does not work with the data type of the parameter specified.

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Windows 64 bit
* Compiler: Visual C++ 2022
* Release or debug mode: Release mode
* Specific flags used (if any): -
* CGAL version: 5.6
* Boost version: the latest

I get the same error message:
1>C:\dev\cgal\Shape_regularization\examples\Shape_regularization\regularize_contour.cpp(22,5): error C2672: 'CGAL::Shape_regularization::Segments::regularize_segments': no matching overloaded function found

If I copy the code from regularize_simple into another project that does not have osqp include & lib. Maybe that lib is missing and is causing your error?

--- a/Shape_regularization/include/CGAL/Shape_regularization/regularize_segments.h
+++ b/Shape_regularization/include/CGAL/Shape_regularization/regularize_segments.h
@@ -70,6 +70,9 @@ namespace Segments {
     that should be addressed. The function is based on the class `QP_regularization`.
     Please refer to that class and these concepts for more information.
 
+    This class requires a `QPSolver` model which defaults to the \ref thirdpartyOSQP "OSQP"
+    library, which must be available on the system.
+
     \tparam InputRange
     a model of `ConstRange` whose iterator type is `RandomAccessIterator`
 

